src/main/java/org/interledger/InterledgerAddress.java

```diff
@@ -88,7 +88,7 @@ default boolean startsWith(final String addressSegment) {
    */
   default boolean startsWith(final InterledgerAddress interledgerAddress) {
     Objects.requireNonNull(interledgerAddress, "interledgerAddress must not be null!");
-    return this.startsWith(interledgerAddress.toString());
+    return this.startsWith(interledgerAddress.getValue());
```

Comment:
Just pushed a few more fixes to this PR -- I ran into a bug with this yesterday where the toString() emits JSON, and so this line never returns `true`. I added unit-test coverage for this condition as well.